sdk/communication/AzureCommunication/Source/Authentication/CommunicationAuthenticationPolicy.swift

```diff
@@ -28,45 +28,33 @@
 import AzureCore
 #endif
 import Foundation
-/**
- The Azure Communication Services authentication policy.
- */
-public class CommunicationUserCredentialPolicy: Authenticating {
-    public var next: PipelineStage?
 
+public class CommunicationPolicyTokenCredential: TokenCredential {    
     private let credential: CommunicationUserCredential
+    var error: AzureError? = nil
 
-    /**
-     Creates a credential policy that authenticates requests using the provided `CommunicationUserCredential`.
-     
-     - Parameter credential: The `CommunicationUserCredential` that will be used to authenticate requests.
-     
-     - SeeAlso: `CommunicationUserCredential.init(...)`
-     */
-    public init(credential: CommunicationUserCredential) {
+    public init(_ credential: CommunicationUserCredential) {
         self.credential = credential
     }
-
-    /**
-     Authenticates an HTTP `PipelineRequest` with a token retrieved from the credential.
-     - Parameters:
-        - request:A `PipelineRequest` object.
-        - completionHandler: A completion handler that forwards the modified pipeline request.
-     */
-    public func authenticate(request: PipelineRequest, completionHandler: @escaping OnRequestCompletionHandler) {
-        credential.token { token, error in
-            if let error = error {
-                completionHandler(request, AzureError.client("Error while retrieving access token", error))
-                return
-            }
-
-            guard let token = token?.token else {
-                completionHandler(request, AzureError.client("Token cannot be empty"))
+    
+    public func token(forScopes scopes: [String], completionHandler: @escaping TokenCompletionHandler) {
+        credential.token { (communicationAccessToken, error) in
+            guard let communicationAccessToken = communicationAccessToken else {
+                self.error = AzureError.client("Communication Token Failure", error)
+                completionHandler(nil, self.error)
                 return
             }
-
-            request.httpRequest.headers[.authorization] = "Bearer \(token)"
-            completionHandler(request, nil)
+            
+            let accessToken = AccessToken(token: communicationAccessToken.token,
+                                          expiresOn: communicationAccessToken.expiresOn)
+            
+            completionHandler(accessToken, nil)
+        }
+    }
+    
+    public func validate() throws {
```

Comment:
When will validate() get called? From my understanding, where there is an error, it will call the completionHandler with the error (line 42-44). Why we need to have a property `error` and return it to caller when validate() is called.  This seems duplication of calling the ` completionHandler(nil, self.error)` in line 44

Reply:
I am not sure when validate is called. I see it is a required function in the protocol. I called the completionHandler with the error cause that should be convention so the client won't just hang there.